external-crates/move/move-compiler/tests/sui_mode/one_time_witness/wrong_field_type_and_extra_field_with_init.exp

```diff
@@ -0,0 +1,27 @@
+error[Sui E02004]: invalid one-time witness declaration
+  ┌─ tests/sui_mode/one_time_witness/wrong_field_type_and_extra_field_with_init.move:2:12
+  │
+2 │     struct BEEP has drop {
+  │            ^^^^ Invalid one-time witness declaration
+3 │         f0: u64,
+  │         -- One-time witness types must have no fields, or exactly one field of type 'bool'
+  ·
+6 │     fun init(_: BEEP, _ctx: &mut sui::tx_context::TxContext) {
+  │                 ---- Used as a one-time witness here
+  │
+  = One-time witness types are structs with the following requirements: their name is the upper-case version of the module's name, they have no fields (or a single boolean field), they have no type parameters, and they have only the 'drop' ability.
+
+error[Sui E02004]: invalid one-time witness declaration
```

Comment:
- Do we want to emit an error for each field? I think one error is enough here, otherwise things could get noisy.
- Is it worth clarifying that the reason this is a problem is because it was passed to `init`?  The current wording kind of assumes the conclusion (i.e. that `BEEP` is an `OTW`) but then contradicts itself, because ...it's not a OTW.

How about something like following wording:

> The first parameter to a two parameter module initializer must be a one-time witness, meaning: its name is the upper-case version of the module's name, it has no fields (or a single boolean field), no type parameters, and only the 'drop' ability.

Reply:
Yeah... I think maybe one message with a note/full description is probably better. This is what I had before and should go back to it. 

> Is it worth clarifying that the reason this is a problem is because it was passed to init?

Isn't that what I do? I point to the usage in init for these cases

Reply:
> Isn't that what I do? I point to the usage in init for these cases

Yes, in terms pointing to the correct location in source, but my comment was about the wording, which read confusingly to me, because it seemed to imply the type both was and wasn't an OTW (the annotation on the source location implied it was an OTW, but then the rules about OTWs and the other annotation suggested otherwise).

It would be valid to say that that is kind of the point (as in, that's the error), but for whatever reason, the softening of "Possible attempted usage..." seems to sit better with me (although, nit, you could drop the "Possible") -- it implies an ordering (e.g. you're in this situation because you attempted this, and that caused a problem here), which follows because the reason a type is considered an OTW is first because it shows up in the `init` parameters, and only subsequently because it meets the criteria.